src/props.ts

```diff
@@ -4,7 +4,7 @@ export interface CommonProps<TData, TReturn> {
   /** The controlled form value to be auto saved   */
   data: TData;
   /** Callback function to save your data */
-  onSave: (data: TData) => Promise<TReturn> | TReturn | void;
+  onSave: (data: TData, isOnUnmount: boolean) => Promise<TReturn> | TReturn | void;
```

Comment:
```suggestion
  onSave: (data: TData, meta: {unmounting: boolean}) => Promise<TReturn> | TReturn | void;
```

Could we wrap this in an object so if we ever need to add more metadata it is not a breaking change to the callback's contract?